test/r401_test.js

```diff
@@ -0,0 +1,336 @@
+  it('should support getting booleans', () => {
+    const pt = patientSource.currentPatient();
+    const immunization = pt.findRecords('Immunization').find(p => p.getId() === '2a986005-b7c9-464c-9da8-0a3220dd8721');
+    expect(immunization.get('primarySource.value')).to.be.true;
+  });
+
+  it('should support getting decimals', () => {
+    const pt = patientSource.currentPatient();
+    const claim = pt.findRecords('Claim').find(p => p.getId() === '58cd648a-5f4d-4306-bef4-49ec64c88c63');
+    expect(claim.get('total.value.value')).to.equal(687.08);
+  });
+
+  it('should support getting integers', () => {
+    const pt = patientSource.currentPatient();
+    const claim = pt.findRecords('Claim').find(p => p.getId() === '58cd648a-5f4d-4306-bef4-49ec64c88c63');
+    expect(claim.get('item')[0].get('sequence.value')).to.equal(1);
+  });
+
+  it('should support getting strings', () => {
+    const pt = patientSource.currentPatient();
+    const procedure = pt.findRecords('Procedure').find(p => p.getId() === '03c48dbb-2e69-451d-877a-b3397a9f3d26');
+    expect(procedure.get('status.value')).to.deep.equal('completed');
+  });
+
+  it('should support getting dateTimes', () => {
+    const pt = patientSource.currentPatient();
+    const medReq = pt.findRecords('MedicationRequest').find(p => p.getId() === '622c5788-3028-41fd-a8cb-164f868d4322');
+    const authoredOn = medReq.getDate('authoredOn.value');
+    expect(authoredOn.isDateTime).to.be.true;
+    expect(authoredOn).to.deep.equal(cql.DateTime.parse('2009-12-01T11:18:29-05:00'));
+  });
+
+  it('should support getting dates', () => {
+    const pt = patientSource.currentPatient();
+    const birthDate = pt.get('birthDate.value');
+    expect(birthDate.isDate).to.be.true;
+    expect(birthDate).to.deep.equal(cql.DateTime.parse('2008-11-06').getDate());
+  });
+
+  it('should support getting times', () => {
+    const pt = patientSource.currentPatient();
+    const observation = pt.findRecords('Observation').find(p => p.getId() === '9c15c801-6bb5-47a7-a9db-8bad0cb6aa68');
+    const valueTime = observation.get('value.value');
+    expect(valueTime.isTime()).to.be.true;
+    expect(valueTime).to.deep.equal(cql.DateTime.parse('0000-01-01T18:23:47.376-05:00').getTime());
+  });
+
+  it('should support getting an option of a choice', () => {
+    const pt = patientSource.currentPatient();
+    const condition = pt.findRecords('Condition').find(p => p.getId() === '9934bc4f-58af-4ecf-bb70-b7cc31987fc5');
+    // In STU3, you use the stub of the choice (e.g., onset[x] datetime is retrieved as onset)
+    expect(condition.get('onset.value')).to.deep.equal(cql.DateTime.parse('2009-08-09T12:18:29-04:00'));
+  });
+
+  it('should support getting an option of a choice using explicit name', () => {
+    // This was needed because the ModelInfo used to indicate MedicationRequest's primaryCodePath as medicationCodeableConcept!
+    // Keep it in case anyone tries to load an older ModelInfo that still does that.
```

Comment:
I don't think the 4.0.1 modelinfo has this legacy, so it's not important for 4.0.1 to support this.  We can probably remove this test.